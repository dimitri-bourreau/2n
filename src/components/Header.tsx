import React, { FunctionComponent, ReactElement } from 'react';
import style from 'styled-components';
import { Trans } from 'react-i18next';

import User from '../interfaces/user';

const HeaderWrapper = style.header`
  padding: 10px 5vw 20px 5vw;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
    background: linear-gradient(37deg, #31419d, #296cb9, #399438);
    background-size: 600% 600%;

    -webkit-animation: AnimationName 46s ease infinite;
    -moz-animation: AnimationName 46s ease infinite;
    animation: AnimationName 46s ease infinite;

  @-webkit-keyframes AnimationName {
      0%{background-position:0% 10%}
      50%{background-position:100% 91%}
      100%{background-position:0% 10%}
  }
  @-moz-keyframes AnimationName {
      0%{background-position:0% 10%}
      50%{background-position:100% 91%}
      100%{background-position:0% 10%}
  }
  @keyframes AnimationName {
      0%{background-position:0% 10%}
      50%{background-position:100% 91%}
      100%{background-position:0% 10%}
  }

  @media all and (min-width: 768px) {
    flex-direction: row;
    padding: 10px 5vw;
  }
`;
const Logo = style.div`
  flex-basis: 7%;
  margin: 10px auto;
  img {
    width: 50px;
  }

  @media all and (min-width: 768px) {
    margin: 0;
  }
`;
const Nav = style.nav`
  flex-basis: 50%;

  @media all and (min-width: 576px) {
    flex-basis: 80%;
  }
  @media all and (min-width: 992px) {
    flex-basis: 70%;
  }
  @media all and (min-width: 1200px) {
    flex-basis: 60%;
  }
`;
const List = style.ul`
  list-style: none;
  padding: 0;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  margin: 0 auto;

  @media all and (min-width: 768px) {
    flex-direction: row;
  }
`;
const ListElt = style.li`
  margin-bottom: 10px;
  a {
    color: white;
    text-transform: uppercase;
    font-weight: 400;
  }

  @media all and (min-width: 768px) {
    margin-bottom: 0;
  }
`;

interface HeaderProps {
  user: User;
}

const Header: FunctionComponent<HeaderProps> = ({
  user,
}: HeaderProps): ReactElement => {
  // const activePath = window.location.pathname.split('').splice(1).join('');

  return (
    <HeaderWrapper>
      <Logo>
        <a href="/" className="img-link">
          <img src="/favicons/android-chrome-512x512.png" alt="logo" />
        </a>
      </Logo>
      <Nav>
        <List>
          <ListElt>
            <a href="/">
              <Trans i18nKey="Header.home">Home</Trans>
            </a>
          </ListElt>
          <ListElt>
            <a href="/ranking">
              <Trans i18nKey="Header.ranking">Ranking</Trans>
            </a>
          </ListElt>
          <ListElt>
            <a href="/more">
              <Trans i18nKey="Header.more">More</Trans>
            </a>
          </ListElt>
          <ListElt>
            <a href="/settings">
              <Trans i18nKey="Header.settings">Settings</Trans>{' '}
              <span role="img" aria-label="settings-and-language" />
              üåê
            </a>
          </ListElt>
          <ListElt>
            <a href={user ? '/me' : '/login'}>
              {user ? (
                <Trans i18nKey="Header.me">My account</Trans>
              ) : (
                <Trans i18nKey="Header.logIn">Log in</Trans>
              )}
            </a>
          </ListElt>
        </List>
      </Nav>
    </HeaderWrapper>
  );
};

export default Header;
