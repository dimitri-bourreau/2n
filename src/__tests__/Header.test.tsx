import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import Header from '../components/Header';

describe('<Header />', () => {
  test('Header component should match snapshot', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Header should render correctly', () => {
    const root = document.createElement('div');
    ReactDOM.render(<Header />, root);
    expect(root.querySelector('#header')).not.toBeNull();
  });
});
