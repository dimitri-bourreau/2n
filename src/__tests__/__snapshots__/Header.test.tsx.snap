// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Header component should match snapshot 1`] = `
<div
  id="header"
>
  <ul
    className="nav nav-tabs"
  >
    <li
      className="nav-item"
    >
      <a
        className="nav-link active"
        href="/"
        onClick={[Function]}
      >
        Game
      </a>
    </li>
    <li
      className="nav-item"
      href="more"
    >
      <a
        className="nav-link"
        href="more"
        onClick={[Function]}
      >
        More
      </a>
    </li>
    <div
      className="ml-auto mr-0 mt-1"
    >
      <div
        id="langDropDown"
      >
        <div
          className="d-table btn-group-sm dropdown"
        >
          <button
            aria-haspopup={true}
            className="btn btn-white"
            onClick={[Function]}
            type="button"
          >
            <span
              aria-label="globle"
              role="img"
            >
              üåê
            </span>
          </button>
          <div
            className="dropdown-menu dropdown-menu-right"
            role="menu"
            tabIndex="-1"
          >
            <button
              className="dropdown-item"
              onClick={[Function]}
              tabIndex="0"
              type="button"
            >
              English
            </button>
            <button
              className="dropdown-item"
              onClick={[Function]}
              tabIndex="0"
              type="button"
            >
              Fran√ßais
            </button>
          </div>
        </div>
      </div>
    </div>
  </ul>
</div>
`;
